### ISSUE TITLE
New typescript project will not build

### ISSUE THREAD

**Issue by grammati** [projen/projen#1788]
Newly generated `typescript` projects will not build.

Steps to reproduce:

1. `mkdir foo`
2. `cd foo`
3. `npx projen new typescript`
4. `npx projen`

Result:

```
% npx projen
👾 default | node .projenrc.js
/Users/chris/projects/foo/.projenrc.js:2
const project = new TypeScriptProject({
                ^

TypeError: TypeScriptProject is not a constructor
    at Object.<anonymous> (/Users/chris/projects/foo/.projenrc.js:2:17)
    at Module._compile (node:internal/modules/cjs/loader:1101:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:17:47
👾 Synthesis failed: Task "default" failed when executing "node .projenrc.js" (cwd: /Users/chris/projects/foo)
```

**Comment by Chriscbr** [comment]
Can you check what version of projen is reported by `npx projen --version`?

Looking at the error, it looks as though the template is file is outdated, new TS projects should be getting generated with:

```js
const { typescript } = require('projen');
const project = new typescript.TypeScriptProject({
```

**Comment by grammati** [comment]
```
% npx projen --version
Need to install the following packages:
  projen
Ok to proceed? (y) y
0.29.17
```

**Comment by grammati** [comment]
Yes, that's a very old version. Looks like a problem with my local node version management. I'll close this, sorry for the false alarm.

**Comment by Chriscbr** [comment]
No worries, yeah that version is quite old. :-)

My understanding is "npx" will try checking your current node_modules folder for an installed version of projen, and if it's not found, it will try checking your parent directory (`../node_modules`), and then its parent (`../../node_modules`), and so on. And if it finds an installed projen, it will use that even if you specified "projen@latest".

**Comment by grammati** [comment]
To file under "today I learned": npx has a cache, and it caches _forever_! The "Need to install... Ok to proceed?" prompt makes this particularly surprising, because it's not actually installing anything - it's pulling from its cache.

After `rm -rf ~/.npm/_npx/*`, I was able to `npx projen` and get the latest version. I _think_ `npx projen@latest` would work too. But without the `@latest` part, npx will use its cached version, no matter how old.

See: https://www.npmjs.com/package/clear-npx-cache
